marklogic-data-hub/src/test/java/com/marklogic/hub/impl/ClearUserModulesTest.java

```diff
@@ -0,0 +1,83 @@
+package com.marklogic.hub.impl;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.marklogic.client.admin.QueryOptionsManager;
+import com.marklogic.client.admin.ResourceExtensionsManager;
+import com.marklogic.client.admin.ServerConfigurationManager;
+import com.marklogic.client.admin.TransformExtensionsManager;
+import com.marklogic.client.io.JacksonHandle;
+import com.marklogic.client.io.QueryOptionsListHandle;
+import com.marklogic.hub.AbstractHubCoreTest;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class ClearUserModulesTest extends AbstractHubCoreTest {
+    ClearUserModulesTest() {
+    }
+
+    @BeforeEach
+    void setUp() throws IOException {
+        Assumptions.assumeTrue(this.isVersionCompatibleWith520Roles());
+        ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+        Path servicesFile = this.getHubProject().getModulesDir().resolve("services");
+        servicesFile.toFile().mkdirs();
+        FileUtils.copyInputStreamToFile(classLoader.getResourceAsStream("ml-modules/services/mlBatches.sjs"), servicesFile.resolve("myResourceExtension.sjs").toFile());
+        Path optionsFile = this.getHubProject().getModulesDir().resolve("options");
+        optionsFile.toFile().mkdirs();
+        FileUtils.copyInputStreamToFile(classLoader.getResourceAsStream("ml-modules-staging/options/default.xml"), optionsFile.resolve("myOptions.xml").toFile());
+        Path transformsFile = this.getHubProject().getModulesDir().resolve("transforms");
+        transformsFile.toFile().mkdirs();
+        FileUtils.copyInputStreamToFile(classLoader.getResourceAsStream("ml-modules/transforms/hubEntitySearchTransform.sjs"), transformsFile.resolve("myTransform.sjs").toFile());
+        this.installProjectInFolder("flow-runner-test");
+    }
+
+    @Test
+    void testClearUserModules() {
+        ServerConfigurationManager configMgr = this.runAsDataHubDeveloper().newFinalClient().newServerConfigManager();
```

Comment:
I think these assertions to verify that each user module type was actually loaded should be part of the setup. Though since there's only one test method, I wouldn't bother with a BeforeEach method. Just make a private method called "installProjectWithEachTypeOfModule" - that documents what's going on here too. That method will use installProjectInFolder and will then verify that each user module exists.